### PR TITLE
Lightbox UI block override tests

### PR DESCRIPTION
**Pull request by artemiomorales**
## What?

After an initial attempt to reenable the lightbox tests in #60150, this PR aims to break that PR down into smaller chunks so that it is easier to review. It includes two UI tests and removes code for many outdated ones.

To put this PR in context: The UI for the lightbox was recently updated in https://github.com/WordPress/gutenberg/pull/57608 with extensive edge cases covered via https://github.com/WordPress/gutenberg/pull/59890#discussion_r1534695371. Now that the UX and implementation are a bit more solid, it makes sense to start reenabling test coverage.

## How?

It adds a test plugin for handling a first set of theme.json settings and some relevant tests.

## Testing Instructions

To run tests locally, use the command `npm run test:e2e -- image.spec.js`

### CODE REVIEW

File: test/e2e/specs/editor/blocks/image.spec.js [hunk 1]
```diff
@@ -865,530 +860,96 @@ test.describe.skip( 'Image - interactivity', () => {
...
-			editor,
-			page,
-			imageBlockUtils,
-		} ) => {
-			const imageBlock = editor.canvas.locator(
-				'role=document[name="Block: Image"i]'
-			);
-			await expect( imageBlock ).toBeVisible();
+		test.beforeAll( async ( { requestUtils } ) => {
+			await requestUtils.deleteAllMedia();
```
Comment (gziolo) [inline test/e2e/specs/editor/blocks/image.spec.js]: There is another beforeAll handler higher in the scope that deletes all media, so one of them should be safe to remove as one of them seem to do nothing.

File: test/e2e/specs/editor/blocks/image.spec.js [hunk 2]
```diff
@@ -865,530 +860,96 @@ test.describe.skip( 'Image - interactivity', () => {
...
+				admin,
 				editor,
 				page,
-				imageBlockUtils,
 			} ) => {
-				const imageBlock = editor.canvas.locator(
-					'role=document[name="Block: Image"i]'
-				);
-				await expect( imageBlock ).toBeVisible();
+				await admin.createNewPost();
```
Comment (gziolo) [inline test/e2e/specs/editor/blocks/image.spec.js]: I see lines where a new post gets created before each test:

https://github.com/WordPress/gutenberg/blob/09298b9fa5f77a4aa10c4d829dba5435acf12d89/test/e2e/specs/editor/blocks/image.spec.js#L855

Is it possible it is no longer necessary?

File: test/e2e/specs/editor/blocks/image.spec.js [hunk 3]
```diff
@@ -865,530 +860,96 @@ test.describe.skip( 'Image - interactivity', () => {
...
 
-			filename = await imageBlockUtils.upload(
-				imageBlock.locator( 'data-testid=form-file-upload-input' )
-			);
-			const image = imageBlock.locator( 'role=img' );
-			await expect( image ).toBeVisible();
-			await expect( image ).toHaveAttribute(
-				'src',
-				new RegExp( filename )
+			uploadedMedia = await requestUtils.uploadMedia(
```
Comment (gziolo) [inline test/e2e/specs/editor/blocks/image.spec.js]: After each test media get deleted:

https://github.com/WordPress/gutenberg/blob/09298b9fa5f77a4aa10c4d829dba5435acf12d89/test/e2e/specs/editor/blocks/image.spec.js#L859L860

Is it possible that second test case is using no longer existing image?

**Review by gziolo — COMMENTED** [review]
This PR has nicely scoped set of changes. I had some questions regarding the lifecycle methods that need to get clarified. Y
Test cases added back and revised make perfect sense and are easy to follow. Nice work!

File: test/e2e/specs/editor/blocks/image.spec.js (hunk 1 above)
Comment (artemiomorales) [reply test/e2e/specs/editor/blocks/image.spec.js]: Ok I removed the extra call

File: test/e2e/specs/editor/blocks/image.spec.js (hunk 3 above)
Comment (artemiomorales) [reply test/e2e/specs/editor/blocks/image.spec.js]: I rearranged the logic so that both tests should be using the same image before it gets deleted.

**Comment by artemiomorales** [comment]
@gziolo Thanks for the feedback and catching the inconsistencies! I've cleaned up the code and rearranged things in the most recent commit. Let me know what you think.

**Review by gziolo — APPROVED** [review]
Looks great, thank you for addressing the feedback.

**Pull request merged**